src/libsyntax/codemap.rs

```diff
@@ -592,8 +592,30 @@ impl CodeMap {
         }
     }
 
-    /// Given a `Span`, try to get a shorter span ending just after the first
-    /// occurrence of `char` `c`.
+    /// Given a `Span`, get a new `Span` covering the first token and all its trailing whitespace or
+    /// the original `Span`.
+    ///
+    /// If `sp` points to `"let mut x"`, then a span pointing at `"let "` will be returned.
+    pub fn span_until_non_whitespace(&self, sp: Span) -> Span {
+        if let Ok(snippet) = self.span_to_snippet(sp) {
+            let mut offset = 0;
+            // get the bytes width of all the non-whitespace characters
+            for c in snippet.chars().take_while(|c| !c.is_whitespace()) {
+                offset += c.len_utf8();
+            }
+            // get the bytes width of all the whitespace characters after that
+            for c in snippet[offset+1..].chars().take_while(|c| c.is_whitespace()) {
```

Comment:
What is the `+1` for?

Reply:
Unless I'm missing something, `!snippet[offset].is_whitespace()`, right? Then `snippet[offset+1].is_whitespace()`. I think.

Reply:
No, `offset` is pointing to the byte immediately after the last non whitespace character.